capstone/fitness/src/main/java/com/google/sps/util/BananaQueue.java

```diff
@@ -0,0 +1,192 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.util;
+
+public class BananaQueue {
+
+  private BananaNode head, foot;
+  private int size;
+
+  public BananaQueue() {
+    head = null;
+    foot = null;
+    size = 0;
+  }
+  
+  /**
+   * Enqueues given PeelNode to PeelQueue associated with tag if PeelNode has a value and the BananaQueue's front element
+   * has a PeelQueue with the same associated name.
+   *
+   * @param peelQueueTag Tag to associated PeelQueue.
+   * @param peel PeelNode to add.
+   */
+  public void enqueuePeel(String peelQueueTag, PeelNode peel) {
+    if(peel == null || head == null || !head.peelQueueExists(peelQueueTag)) {
+      return;
```

Comment:
better to return a boolean to indicate if the operation is successful

Reply:
done